src/material/chips/chip-icons.ts

```diff
@@ -42,6 +42,55 @@ export class MatChipTrailingIcon extends MatChipAction {
   override _isPrimary = false;
 }
 
+/**
+ * Directive to edit the parent chip when the leading action icon is clicked or
+ * when the ENTER key is pressed on it.
+ *
+ * Recommended for use with the Material Design "edit" icon
+ * available at https://material.io/icons/#ic_edit.
+ *
+ * Example:
+ *
+ * ```
+ * <mat-chip>
+ *   <button matChipEdit aria-label="Edit">
+ *     <mat-icon>edit</mat-icon>
+ *   </button>
+ * </mat-chip>
+ * ```
+ */
+
+@Directive({
+  selector: '[matChipEdit]',
+  host: {
+    'class':
+      'mat-mdc-chip-edit mat-mdc-chip-avatar mat-focus-indicator ' +
+      'mdc-evolution-chip__icon mdc-evolution-chip__icon--primary',
+    'role': 'button',
+    '[attr.aria-hidden]': 'null',
+  },
+  providers: [{provide: MAT_CHIP_EDIT, useExisting: MatChipEdit}],
+})
+export class MatChipEdit extends MatChipAction {
+  override _isPrimary = false;
+
+  override _handleClick(event: MouseEvent): void {
+    if (!this.disabled) {
+      event.stopPropagation();
+      event.preventDefault();
+      this._parentChip._edit();
+    }
+  }
+
+  override _handleKeydown(event: KeyboardEvent) {
+    if ((event.keyCode === ENTER || event.keyCode === SPACE) && !this.disabled) {
+      event.stopPropagation();
+      event.preventDefault();
+      this._parentChip._edit();
```

Comment:
This appears to allow editing even if the chip isn't marked as editable.

Reply:
That's correct. Editable controls the double-click and focus-enter, the button logic is independent.

Reply:
I suspect that may be confusing for developers using the component since the only way left to conditionally disable editing is by completely removing the edit icon.